test/service/test_manifest_async.py

```diff
@@ -310,11 +311,75 @@ def test(self, mock_uuid, mock_helper):
                             self.assertEqual(object_url, response.headers['Location'])
 
     params = {
-        'token': Token(execution_id='7c88cc29-91c6-4712-880f-e4783e2a4d9e',
+        'token': Token(execution_id=object_key + '_0',
                        request_index=0,
                        wait_time=0).encode()
     }
 
+    @patch_step_function_helper
+    def test_idempotent_manifest_request(self, step_function_helper):
+        """
+        Multiple request for the same manifest shouldn't trigger multiple
+        step function executions.
+        """
+        step_function_helper.start_execution.side_effect = ClientError({
+            'Error': {
+                'Code': 'ExecutionAlreadyExists'
+            }
+        }, '')
+        step_function_helper.describe_execution.return_value = {
+            'status': 'RUNNING'
+        }
+        url = self.base_url.set(path='/fetch/manifest/files')
+        location = set()
+        status = set()
+        for i in range(3):
+            response = requests.get(str(url)).json()
+            location.add(response['Location'])
+            status.add(response['Status'])
+        url.args.add('token',
+                     Token(execution_id=self.object_key + '_0',
+                           request_index=1,
+                           wait_time=1).encode())
+        self.assertEqual(str(url), one(location))
+        self.assertEqual(301, one(status))
+
+    @patch_step_function_helper
+    def test_idempotent_manifest_request_retry(self, step_function_helper):
+        """
+        Multiple request for the same manifest shouldn't trigger multiple
+        step function executions.
+        """
+        step_function_helper.start_execution.side_effect = [ClientError({
+            'Error': {
+                'Code': 'ExecutionAlreadyExists'
+            }
+        }, ''), 0]
```

Comment:
I find these `ClientError` instantiations to be consistently ugly, repetitive, and verbose, and the adjacency with `0` makes this very difficult to read. I tried making adjustments to the wrapping but it was a marginal improvement at best. I think it might be a good idea to extract most of this to a method, probably in a separate commit titled something like `Consolidate ClientError  boilerplate in test_manifest_async` or something like that:

```
    def _client_error(self, code: str) -> ClientError:
         return ClientError({
             'Error': {
                 'Code': code
             }
         }, '')
```

And then:

```suggestion
        step_function_helper.start_execution.side_effect = [
             self._client_error('ExecutionDoesNotExist'),
             0
        ]
```

becomes much nicer.